### ISSUE TITLE
Fix readthedocs config to stop warning that latest branch is a development branch.

### ISSUE THREAD

**Issue by pushfoo** [pythonarcade/arcade#2404]
## Bug Report

Readthedocs now injects "helpful" JS that misleads users:
![image](https://github.com/user-attachments/assets/f2faad5b-dfb7-4fe4-bde8-499812750fbd)

We need to gain control of that.

**Comment by DragonMoffon** [comment]
It also warns that the development page is out of date

**Comment by pushfoo** [comment]
**TL;DR:** RTD switched on some plugins as defaults on October 7th, but I found the off switch

### Who can fix this?

Anyone with access to the new plugins control panel: @einarf, @Cleptomania, or @pvcraven.

### How can they fix this?
#### Overview
| Option | What | Why it could be     worth it |
|----------|------------------|--|
| The Good | Turn it off through the control panel | Ship 3.0 fast |
| The Bad    | Re-use their non-stable JS API | It's still injected when a project switches off the plugin |
| The Ugly   | Fuss with version number regex | It may be the best long-term option |

#### 1. The Good: turn it off fast
This'll let us move on with other 3.0-critical tasks ASAP:

1. Go to https://app.readthedocs.org/dashboard/
3. Click the Arcade project
4. Look one row down below the title
5. Follow it right until you see a "settings" button:
    ![image](https://github.com/user-attachments/assets/3374bae2-3864-46fa-a3c5-037cc983e4e2)
6. Click it
7. In the left-hand sidebar, click the `Addons` option:
    ![image](https://github.com/user-attachments/assets/7fce06fd-f764-4078-8033-6509a845547f)
8. Look to the box to the right of left-hand sidebar to find the `Notifications` tab: 
    ![image](https://github.com/user-attachments/assets/b33f2158-3eb4-4bf4-8621-8e89f914f5ff)
10. Click it
11. Uncheck this box 
    ![image](https://github.com/user-attachments/assets/9d09fd40-0773-4cda-b512-48e4b7acf112)
12. Hit save

:crossed_fingers: Hopefully, that should be it.

#### 2. Bad: re-use their JS
They inject [`notifications.js`](https://github.com/readthedocs/addons/blob/main/src/notification.js) even when a project disables version notifications in the add-ons panel. The caveat is that this probably isn't meant to be public API:
1. There's no real doc for this in the main Readthedocs documentation
2. There's no doc for it on the addons repo (see https://github.com/readthedocs/addons/)

For an example of how how the inject it:

1. Go to https://better-sum.readthedocs.io/en/development/ (I switched the add-on off in the control panel as described above)
2. Open your dev tools
3. Go to the sources tab
4. Go to `top` > `readthedocs` > `addons` > `src` > `notifcation.js`
5. See the [`NotificationElement` class](https://github.com/readthedocs/addons/blob/895626d866cf2323a99a4c4814a0ec0397ccb49c/src/notification.js#L14) and [`calculateStableLatestVersionWarning()`](https://github.com/readthedocs/addons/blob/895626d866cf2323a99a4c4814a0ec0397ccb49c/src/notification.js#L205)

#### 3. Ugly: Mess with regex
This might be "right" solution in the long term:
1. See https://github.com/mbarkhau/bumpver#pattern-examples
2. Also read the above [Normalization Caveats](https://github.com/mbarkhau/bumpver?tab=readme-ov-file#normalization-caveats) section on  funny version names
3. Mess with our config to get it working

### Why is this happening?

A blog post from July announced an upcoming auto-enable [here](https://about.readthedocs.com/blog/2024/07/addons-by-default/) of new plugins for October 7th, about the time we saw the issue arise:
> We are announcing the introduction of [Read the Docs Addons](https://about.readthedocs.com/blog/2024/04/enable-beta-addons/) to all the projects by default starting on October 7, 2024. Read the Docs Addons is the new modular implementation of all the features Read the Docs adds on top of your documentation: analytics, flyout menu, notifications and search, among others.

I found it via their new documentation [version warning notifications](https://docs.readthedocs.io/en/stable/versions.html#version-warning-notifications).

**Comment by Cleptomania** [comment]
This is disabled on read the docs now, and seems to have gone away(might take a little bit of time for some people due to JS caches or something)
